tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrderingPartyControllerTests.cs

```diff
@@ -113,78 +121,85 @@ public static void Constructor_VerifyGuardClauses()
 
-            await controller.UpdateAsync(callOffId, model);
+            service.Setup(o => o.GetOrder(callOffId)).ReturnsAsync(order);
+            service.Setup(o => o.SetOrderingParty(order, It.IsAny<OrderingParty>(), It.IsAny<Contact>())).Callback(() =>
+            {
+                order.OrderingParty.Name = model.Name;
+                order.OrderingParty.OdsCode = model.OdsCode;
+            });
+            contactDetailsService.Setup(o => o.AddOrUpdatePrimaryContact(
+                    It.Is<Contact>(c => c == originalContact),
+                    It.Is<PrimaryContactModel>(c => c == model.PrimaryContact)))
+                .Returns(order.OrderingPartyContact);
 
+            await controller.UpdateAsync(callOffId, model);
+            service.Verify(o => o.GetOrder(callOffId));
+            service.Verify(o => o.SetOrderingParty(order, It.IsAny<OrderingParty>(), It.IsAny<Contact>()));
             contactDetailsService.Verify(s => s.AddOrUpdatePrimaryContact(
                 It.Is<Contact>(c => c == originalContact),
                 It.Is<PrimaryContactModel>(c => c == model.PrimaryContact)));
-        }
-
-        [Test]
-        [InMemoryDbAutoData]
-        public static async Task UpdateAsync_SavesToDb(
-            [Frozen] ApplicationDbContext context,
-            [Frozen] CallOffId callOffId,
-            Order order,
-            OrderingPartyModel model,
-            OrderingPartyController controller)
-        {
-            context.Order.Add(order);
-            await context.SaveChangesAsync();
-
-            await controller.UpdateAsync(callOffId, model);
 
-            context.Set<Order>().First(o => o.Equals(order)).OrderingParty.Should().BeEquivalentTo(
-                model,
-                o => o.Including(p => p.Name).Including(p => p.OdsCode));
+            order.OrderingPartyContact.Should().BeEquivalentTo(originalContact);
         }
 
         [Test]
         [InMemoryDbAutoData]
         public static async Task UpdateAsync_SuccessfulUpdate_ReturnsNoContentResult(
-            [Frozen] ApplicationDbContext context,
+            [Frozen] Mock<IOrderingPartyService> service,
             [Frozen] CallOffId callOffId,
             Order order,
             OrderingPartyModel model,
             OrderingPartyController controller)
         {
-            context.Order.Add(order);
-            await context.SaveChangesAsync();
+            service.Setup(o => o.GetOrder(callOffId)).ReturnsAsync(order);
+            service.Setup(o => o.SetOrderingParty(order, It.IsAny<OrderingParty>(), It.IsAny<Contact>())).Callback(() =>
+            {
+                order.OrderingParty.Name = model.Name;
+                order.OrderingParty.OdsCode = model.OdsCode;
+            });
 
             var result = await controller.UpdateAsync(callOffId, model);
+            service.Verify(o => o.GetOrder(callOffId));
+            service.Verify(o => o.SetOrderingParty(order, It.IsAny<OrderingParty>(), It.IsAny<Contact>()));
```

Comment:
I don't think we need these additional `Verify` tests here as these are covered as part of the test.

However, If you think it's worthwhile verifying somewhere that `UpdateAsync` invokes `GetOrder` and `SetOrderingParty`, I'd recommend adding two separate tests for that.

Reply:
Removed `service.Verify()`
And added new test method for `SetOrderingParty` invoke